Enable visual snippets for vim (#359)
artisticat1/obsidian-latex-suite: DOCS.md

But this intro can't do it justice, so for unfamiliar that are interested, it's recommended to read [Vims](https://vimhelp.org/usr_02.txt.html#usr_02.txt) for a proper explanation.

Obsidian uses a subset of vim's keybindings maintained by [this repo](https://github.com/replit/codemirror-vim). This subset does not cover everything like [select-mode](https://vimhelp.org/visual.txt.html#Select-mode) and has its own bugs. For any issues that may arise, please first check if the problem lies with [CodeMirror-Vim](https://github.com/replit/codemirror-vim) at [vim-playground](https://raw.githack.com/replit/codemirror-vim/master/dev/web-demo.html) or Obsidian before reporting it here.

### Select mode
You can trigger visual snippets in insert mode, but not in visual mode.
Because of this, a shortcut in the vim extension is defined that allows you to switch between insert mode and visual mode while keeping the selection. This should also work with multicursor.

You can define this shortcut in the settings by enabling vim mode in obsidian and in this plugin.

> The shortcut should be a vim keybinding with no spaces. Some keybindings are not supported by the original vim extension. Due to this please first try seeing if the shortcut works for other keys like `:imap <C-a> <Esc>` or `:vmap <A-h> 0`. It is known that the shift key like `:map <S-a> <Esc>` does not work.

Reviewer: ```suggestion
The shortcut should be a vim keybinding with no spaces. Some keybindings are not supported by the original vim extension. Due to this, please first see if the shortcut works for other keys, such as `:imap <C-a> <Esc>` or `:vmap <A-h> 0`. It is known that the shift key (`:map <S-a> <Esc>`) does not work.
```

The shortcut from visual mode to select mode only works in visual mode and the shortcut from select mode to visual mode only works when something is selected and you are in insert mode.

> Macros are not yet supported for visual snippets

Reviewer: ```suggestion
Macros are not yet supported for visual snippets.
```
